src/main/java/org/broadinstitute/hellbender/tools/funcotator/mafOutput/MafOutputRenderer.java

```diff
@@ -0,0 +1,733 @@
+        }
+    }
+
+    @Override
+    public void close() {
+        printWriter.flush();
+        printWriter.close();
+    }
+
+    @Override
+    public void write(final VariantContext variant, final List<Funcotation> funcotations) {
+
+        // Loop through each alt allele in our variant:
+        for ( final Allele altAllele : variant.getAlternateAlleles() ) {
+
+            // Create our output maps:
+            final LinkedHashMap<String, Object> outputMap = new LinkedHashMap<>(defaultMap);
+            final LinkedHashMap<String, Object> extraFieldOutputMap = new LinkedHashMap<>();
+
+            // Get our funcotations for this allele and add them to the output maps:
+            for ( final Funcotation funcotation : funcotations ) {
+                if ( funcotation.getAltAllele().equals(altAllele) ) {
+                    // Add all the fields from the other funcotations into the extra field output:
+                    for ( final String field : funcotation.getFieldNames() ) {
+                        setField(extraFieldOutputMap, field, funcotation.getField(field));
+                    }
+                }
+            }
+
+            // Now add in our annotation overrides so they can be aliased correctly with the outputFieldNameMap:
+            extraFieldOutputMap.putAll(overrideAnnotations);
+
+            // Go through all output fields and see if any of the names in the value list are in our extraFieldOutputMap.
+            // For any that match, we remove them from our extraFieldOutputMap and add them to the outputMap with the
+            // correct key.
+            for ( final Map.Entry<String, List<String>> entry : outputFieldNameMap.entrySet() ) {
+                for ( final String fieldName : entry.getValue() ) {
+                    if ( extraFieldOutputMap.containsKey(fieldName) ) {
+                        outputMap.put(entry.getKey(), extraFieldOutputMap.remove(fieldName));
+                        break;
+                    }
+                }
+            }
+
+            // Merge our output maps together:
+            outputMap.putAll(extraFieldOutputMap);
+
+            // Now translate fields to the field names that MAF likes:
+            final LinkedHashMap<String, String> mafCompliantOutputMap = replaceFuncotationValuesWithMafCompliantValues(outputMap);
+
+            // Write our header if we have to:
+            if ( ! hasWrittenHeader ) {
+                writeHeader(mafCompliantOutputMap);
+            }
+
+            // Write the output (with manual annotations at the end):
+            writeLine(
+                    mafCompliantOutputMap.values().stream().collect(Collectors.joining(MafOutputRendererConstants.FIELD_DELIMITER))
+                    + manualAnnotationSerializedString
+            );
+        }
+    }
+
+
+    //==================================================================================================================
+    // Static Methods:
```

Comment:
Sounds good.  I generally like to leave them in so that when I create methods / fields that go in the blocks there is an obvious place in which to put them.